CarpetX/src/driver.cxx

```diff
@@ -2575,6 +2581,32 @@ CCTK_INT CarpetX_GetBoundarySizesAndTypes(
   return 0;
 }
 
+const int *ArrayGroupSizeB(const cGH *cctkGH, int dir, int gi,
+                           const char *groupname) {
+  if (groupname) {
+    gi = CCTK_GroupIndex(groupname);
+  }
+  assert(gi >= 0 and gi < CCTK_NumGroups());
+
+  // TODO: handle case of no storage allocated
+
+  cGroup group;
+  int ierr = CCTK_GroupData(gi, &group);
+  assert(!ierr);
+  assert(dir >= 0 && dir < group.dim);
+  if (group.grouptype == CCTK_GF) {
```

Comment:
I have a slight preference for a `switch` statement here.

Reply:
Done